main.py

```diff
@@ -6,80 +6,138 @@
     return response.json()
 
+
 def is_valid_proxy(proxy):
-    """Check if the proxy is valid and not from Russia (youtube is slowed down there)."""
-    return proxy["host"] and proxy["country"] != "Russia"
+    """Check if the proxy is valid."""
+    return proxy.get("host") is not None and proxy.get("country") != "Russia"
+
 
 def construct_proxy_string(proxy):
     """Construct a proxy string from the proxy dictionary."""
     if proxy.get("username"):
-        return f'{proxy["username"]}:{proxy["password"]}@{proxy["host"]}:{proxy["port"]}'
+        return (
+            f'{proxy["username"]}:{proxy["password"]}@{proxy["host"]}:{proxy["port"]}'
+        )
     return f'{proxy["host"]}:{proxy["port"]}'
 
+
 def test_proxy(proxy):
     """Test the proxy by measuring the download time."""
     proxy_str = construct_proxy_string(proxy)
-    print(f'Testing {proxy_str}')
+    print(f"Testing {proxy_str}")
 
     start_time = time.perf_counter()
     try:
-        response = requests.get(SPEEDTEST_URL, stream=True, proxies={"http": proxy_str}, timeout=5)
+        response = requests.get(
+            SPEEDTEST_URL,
+            stream=True,
+            proxies={"http": f"http://{proxy_str}"},
+            timeout=5,
+        )
         response.raise_for_status()  # Ensure we raise an error for bad responses
 
-        total_length = response.headers.get('content-length')
+        total_length = response.headers.get("content-length")
         if total_length is None or int(total_length) != 5242880:
             print("No content or unexpected content size.")
             return None
 
         with io.BytesIO() as f:
-            download_time, downloaded_bytes = download_with_progress(response, f, total_length, start_time)
+            download_time, _ = download_with_progress(
+                response, f, total_length, start_time
+            )
             return {"time": download_time, **proxy}  # Include original proxy info
     except requests.RequestException:
         print("Proxy is dead, skipping...")
         return None
 
+
 def download_with_progress(response, f, total_length, start_time):
     """Download content from the response with progress tracking."""
     downloaded_bytes = 0
     for chunk in response.iter_content(1024):
         downloaded_bytes += len(chunk)
         f.write(chunk)
         done = int(30 * downloaded_bytes / int(total_length))
-        if done > 3 and (downloaded_bytes//(time.perf_counter() - start_time) / 100000) < 1.0:
+        if (
+            done > 3
+            and (downloaded_bytes // (time.perf_counter() - start_time) / 100000) < 1.0
+        ):
             print("\nProxy is too slow, skipping...")
-            return float('inf'), downloaded_bytes
-        sys.stdout.write("\r[%s%s] %s Mbps" % ('=' * done, ' ' * (30-done), downloaded_bytes//(time.perf_counter() -
-        start_time) / 100000))
+            return float("inf"), downloaded_bytes
+        if done > 5:
+            break
```

Comment:
Lines 84 and 85 might need to be above line 78, because a value i.e. 6 will into the above block of code and never reach the break.

Reply:
Fixed, thank you :)